### PR TITLE
feat: add color picker app [EXT-4049]

### PR DESCRIPTION
**Pull request by fabe**
Adds the color picker app. `package.json` probably needs some love for the deployment to work?

Preview

### Demo

[Install to Contentful](https://app.contentful.com/deeplink?link=apps&id=6PbEUk8SPxj1hFMO6ttQo)

### CODE REVIEW

File: apps/color-picker/src/index.css [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  box-shadow: inset 0 0 0 1px rgba(0, 0, 0, 0.1);
+}
+
+input[type='color'] {
+  -webkit-appearance: none;
+  border: none;
+  width: 32px;
+  height: 32px;
+  border-radius: 6px;
+  background: transparent;
```
Comment (loweisz) [inline apps/color-picker/src/index.css:34]: Why not use emotion for these styles as well?

File: apps/color-picker/src/locations/ConfigScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+      themes: [
+        {
+          ...parameters.themes[0],
+          colors: [
+            ...parameters.themes[0].colors,
+            { id: uuidv4(), name: '', value: '#0088cc' },
+          ],
+        },
+      ],
+    });
```
Comment (loweisz) [inline apps/color-picker/src/locations/ConfigScreen.tsx:50]: What about extracting the "swatch part" as a new state and then only on saving add it to parameters? Then you don't need to do this spreading all the time?

File: apps/color-picker/src/locations/ConfigScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+  };
+
+  const updateSwatch = (swatch: Color) => {
+    const index = parameters.themes[0].colors.findIndex(
+      (i) => i.id === swatch.id
+    );
+    let newTheme = parameters.themes[0];
```
Comment (loweisz) [inline apps/color-picker/src/locations/ConfigScreen.tsx:69]: nit: doesn't need to be `let` when you only change parameters

File: apps/color-picker/src/locations/ConfigScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+          colors: parameters.themes[0].colors.filter((i) => i.id !== swatch.id),
+        },
+      ],
+    });
+  };
+
+  const updateSwatch = (swatch: Color) => {
+    const index = parameters.themes[0].colors.findIndex(
+      (i) => i.id === swatch.id
+    );
```
Comment (loweisz) [inline apps/color-picker/src/locations/ConfigScreen.tsx:68]: just for sanity, I would add a -1 check, to make it is actually found

File: apps/color-picker/src/locations/Field.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            <Flex alignItems="center" gap="spacingXs">
+              <span
+                css={css`
+                  width: 16px;
+                  height: 16px;
+                  box-shadow: inset 0 0 0 1px rgba(0, 0, 0, 0.1);
+                  background-color: ${value?.value || '#ffffff'};
+                  display: flex;
+                  border-radius: 4px;
+                `}
```
Comment (loweisz) [inline apps/color-picker/src/locations/Field.tsx:63]: In general, I think moving all the CSS styles into a dedicated style file or at least outside of the HTML would make it a bit easier to read 😬

File: apps/color-picker/src/locations/Field.tsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+              />
+              <span
+                css={css`
+                  display: flex;
+                  gap: ${tokens.spacing2Xs};
+                `}
+              >
```
Comment (loweisz) [inline apps/color-picker/src/locations/Field.tsx:70]: You can just use the `Flex` component here: 
`<Flex gap="spacing2Xs" >`

File: apps/color-picker/src/locations/Field.tsx [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+            >
+              <Flex alignItems="center" gap="spacingXs">
+                <span className={styles.colorBox(color.value)} />
+                {color.name}
+              </Flex>
+            </Menu.Item>
+          ))}
+          {allowCustomValue && (
+            <Menu.Item onClick={() => customColorPicker?.current?.click()}>
+              Other...
```
Comment (andipaetzold) [inline apps/color-picker/src/locations/Field.tsx:126]: Should this be "Custom" as we are selecting a custom value

File: apps/color-picker/src/types.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+export interface Theme {
+  id: string;
+  name: string;
+  colors: Color[];
+}
+
+export interface AppInstallationParameters {
+  themes: Theme[];
```
Comment (andipaetzold) [inline apps/color-picker/src/types.ts:14]: We only use one theme at the moment. Should we still store an array here? Any plans on multiple themes?

File: apps/color-picker/src/locations/Field.tsx (hunk 7 above)
Comment (fabe) [reply apps/color-picker/src/locations/Field.tsx:126]: Yes, makes sense!

File: apps/color-picker/src/locations/Field.tsx [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+};
+
+interface MenuButtonProps {
+  showChevron?: boolean;
+  name: string;
+  value?: FieldValue;
+  onClick?: () => void;
+}
+
+function MenuButton({ showChevron, name, value, onClick }: MenuButtonProps) {
```
Comment (loweisz) [inline apps/color-picker/src/locations/Field.tsx:148]: Maybe worth moving this into a new file?

**Review by loweisz — APPROVED** [review]
Code looks good to me

**Pull request merged**